From patchwork Mon May  2 06:45:51 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michael Riesch <michael.riesch@wolfvision.net>
X-Patchwork-Id: 12833892
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 68731C433F5
	for <linux-rockchip@archiver.kernel.org>;
 Mon,  2 May 2022 06:46:32 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:Cc
	:To:From:Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:
	List-Owner; bh=SEbV8ovAkSG3CF0QXBdI2RuUo44H/Per77efOnhPQUo=; b=ZOvoFynuSKUUj4
	tQFJhm36itSv1WXpmmpTX4Gtr9cKxcZqTAgdJTkHf2TMNOSuIpQKd7YncW9JZ4pR3Tx/rR9bPaLcD
	TxXT/ZMv4oGAtI843Lse61h8m+Q0EM/3ZwAM11jg48aQqQsbJUxINQa4i3Z1gVrZF/+E+WtsS458R
	meZwBpuqUfD1TbCUlqfGHyNiTibUPIRo32nGk1GzkPxvjSk2b6Dlih+m/PiHTeAs4yDXOulgCgQ6D
	BLRIOnCFJxvqGh2IZhlG3DUnFAH4RJqVRhzNIXhIY4Wf9KZ5WSvKKBAPanRFnzxbzlyuDrhejfD7z
	JuhJBcpeW+MGRe3+EOvg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nlPpB-0004pz-5B; Mon, 02 May 2022 06:46:25 +0000
Received: from mail-db8eur05on2062c.outbound.protection.outlook.com
 ([2a01:111:f400:7e1a::62c]
 helo=EUR05-DB8-obe.outbound.protection.outlook.com)
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1nlPox-0004ml-UR; Mon, 02 May 2022 06:46:15 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=a8of7JKJ8WajbO7iEaQdEJtnPaET27D73gG7glYcbyHeqOCMiIQtZdbIOVLEd5TFzvLBcSzQht4pL3nlYQ/neg6qhBLhSazRJBXhncCxb/vA1nHq8AT3h1Xb+8ZcHbeoMcE0OJ1hJ8Me1U8ZPhdni3UKnqr5u7taajowBh2htn8ZHibBK54C9LhXn8d0PY5U5GTZWPIDs0rk3xez7dWFpDkLPUHNA8OcH0FNtEH0AqNIi5V+6GO4DNPMmaEnw3WelSy+2SeqxR23rHcwUcANrgbx9WvViD/NLKGI6s0gBmvroVBLFRXn+jbdRTMbjUYtlf3f9Wqeo1zMFhz0aHy7uw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=YcHxIlP47SjL8DUajpla9eRJDZIVYLstav6t1q7csS4=;
 b=XlHs5r9bs6+Ccqc2uMmuozm2NBQsF55pv1hvzDk5PwIFWRVCEnDYLzRfN9QlTJolvvwetHj8WAK2EMFOdKH0/8PYBnybIfK+OQPg5JLqAAi3b5oTIRd5xKO8m/sG0sbzhsVfkh+pDEa3I0A6nCzyRS/Ad/HqwAJTXmRxqvYz+fmCTaf2xNB51mmPdqgGlNMnrY0/Z9441929vHL4/iH+q0tmZcW8pNEhtC9vKT9fbKYQVJUdsIzgWNaC0Jws08oEiSNy/REdNi/xthWSwwMqDxdztUPs5r046aR4HKEEmdPactoBoIH7r3w3Fv+mfx/Bh1TR0+pCie8Sx8ELh+OTBg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=wolfvision.net; dmarc=pass action=none
 header.from=wolfvision.net; dkim=pass header.d=wolfvision.net; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=wolfvision.net;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=YcHxIlP47SjL8DUajpla9eRJDZIVYLstav6t1q7csS4=;
 b=lSGLY3J/tPONK9F3atb3TnHrCYS5uVkgyr1g0aVlaNTevKBtvyWeRPPGzblu7h7CAW1104oy4ZanwyHDF0xmI6o2o6AMbo23eDL2sh5ilZxga+/IhZqERdGNAKVNOXIfXsf6t0LbNP2xLCMIgak/DSIj80puuNe4Mp7Ft8EMXu4=
Authentication-Results: dkim=none (message not signed)
 header.d=none;dmarc=none action=none header.from=wolfvision.net;
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com (2603:10a6:3:e1::23)
 by AS8PR08MB6407.eurprd08.prod.outlook.com (2603:10a6:20b:33e::16)
 with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5186.15; Mon, 2 May
 2022 06:46:03 +0000
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::3de2:62a:3915:9d6d]) by HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::3de2:62a:3915:9d6d%7]) with mapi id 15.20.5186.028; Mon, 2 May 2022
 06:46:02 +0000
From: Michael Riesch <michael.riesch@wolfvision.net>
To: devicetree@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-rockchip@lists.infradead.org, linux-kernel@vger.kernel.org
Cc: Rob Herring <robh+dt@kernel.org>,
 Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Heiko Stuebner <heiko@sntech.de>,
 Michael Riesch <michael.riesch@wolfvision.net>,
 Liang Chen <cl@rock-chips.com>, Peter Geis <pgwipeout@gmail.com>,
 Nicolas Frattaroli <frattaroli.nicolas@gmail.com>,
 Frank Wunderlich <frank-w@public-files.de>
Subject: [PATCH v2] arm64: dts: rockchip: enable otg/drd operation of
 usb_host0_xhci in rk3568
Date: Mon,  2 May 2022 08:45:51 +0200
Message-Id: <20220502064551.202613-1-michael.riesch@wolfvision.net>
X-Mailer: git-send-email 2.30.2
X-ClientProxiedBy: ZR0P278CA0144.CHEP278.PROD.OUTLOOK.COM
 (2603:10a6:910:40::23) To HE1PR0802MB2426.eurprd08.prod.outlook.com
 (2603:10a6:3:e1::23)
MIME-Version: 1.0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 76a4449e-b62e-4c90-3089-08da2c076c62
X-MS-TrafficTypeDiagnostic: AS8PR08MB6407:EE_
X-Microsoft-Antispam-PRVS: 
 <AS8PR08MB6407A8AA10546C6CEE36B60AF2C19@AS8PR08MB6407.eurprd08.prod.outlook.com>
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 eiWqEH4FEk6QCrWuhNOs3CobnOtPto4GXsId+GtXrZeKtiPNJ6qPG5sxsEXAh3KS7diTuy8/MJvFhUk6TPKqn/9ztn2SnlluhUb+OTOLvjh2A5Yo61XQBha/Xj+UX6Pp7BbTHqC0wE5vysfwxOcPfBJTMCIuzeehY6HiSyz0WKNYf/DSZWBe5zyAj8ChRC9lf2D3qnEWFYErVB9qUXnztFgZioc//LYvWMgMqxxtP1WDvVc7BbDx3FwL+WV1Uni0eTdnrpxRmFHIGcmnerYVz3ddIsyK8Ut31xOP7BSMvNqJ5l8Bn+6VPlQBJgRXHpT8PMAuvXh6GOTA9lV2xL0z9+iC5M3VNvKwGpwU3CLB5pkq9pmjGES2z+jJW6gxveHDvfDFshwxrDrbf+zJKmHQYRE1hTZIgoFxmQdFv5v22Gj2BFQiGO4W5VP+NUfAC4XzagSLQK1yjboGIDrUNXKfub/FG1Es1ytZKTyPS2VVXy51GLlqJ8kxeEi1xkBpD53MOw6tJyuNORY/FR9HLjbbQAALLlC2Hl7Df1lyz9sDQTTjTnBTIQm+71LH9UsZ3wHSJHcMaz5f6s01M2uWFOQoVxT5Pc8iCcxZtwrzQi4Ql8Xm5L/cGt9yWSWkeycisMVuFOtw2ReKAohk61tCFpZhqw==
X-Forefront-Antispam-Report: CIP:255.255.255.255; CTRY:; LANG:en; SCL:1; SRV:;
 IPV:NLI; SFV:NSPM; H:HE1PR0802MB2426.eurprd08.prod.outlook.com; PTR:;
 CAT:NONE;
 SFS:(13230001)(4636009)(346002)(376002)(136003)(39830400003)(366004)(396003)(2616005)(54906003)(508600001)(6486002)(6506007)(6512007)(52116002)(186003)(1076003)(86362001)(36756003)(66476007)(66556008)(66946007)(4326008)(8676002)(2906002)(44832011)(8936002)(5660300002)(4744005)(7416002)(38100700002)(316002)(6666004);
 DIR:OUT; SFP:1101;
X-MS-Exchange-AntiSpam-MessageData-ChunkCount: 2
X-MS-Exchange-AntiSpam-MessageData-0: 
 tms4058hipIzToqYeWaumFXGfUS6RMoh03N3jUT4MMlLkjPCQW3l3DtzSbvDzCvpaXvy7GNyA3yK5DlA2P8a1OqnzMuiQFQncf6oeW3PwbXDs5+4BXQrOvzfcE7arofZp7gsInvRnW0LYRUhhvi8YqXbf5sTa0FkU8BpBO4tiI91EekU8dm8VhitvT/eLY0peGJNx7ppwluRux4FeTM8a0rb/ce7MPBysV0vQC4Ow5u0/csi6gzkQ1PH3ZXlwW3w8TaTc75eX4v6zld0UoKrvSMBe7YuCAIl7lZMSfNOWVXbJcWWPE6QaLMUwpBirL/krOlUm6D16vbUtPCPZ0+vyHDGNQx9TOdXLmfJsDn5EdljshjAVbeC8ISFWspvQl14kwoUa5/McsmH4KhVfp4ndqq415b808mEQHFU8qDom2G7VXUyGHyor9TcwkqwzG96apD8t0/nVbEVzmVSzRk6sQZ1MGbj1Geo+T83oPrsBtZ+uBnQTu+q2rj6+BLd7iNug4NJodtDgS13DSLheVN4M9V1cZAFwdhlhvjQalIejJStOX7deI6RcJcW5M6qinm0ToAzGrGrfNGcKgClUE0ydMDVfOqmkwS4cL0QlgD6QX31z+UZQezvgVPEPXDE+UDdkswFcUAywrCMJZURUV39iUtvNnyJPfij3ifJWGhz7YhTfMIz5aEm8xKHY9StJaZ+0O96kNlJ3VDUcmuuYPouWun/nYe1o4dZQvEhy3uFRBXG7GJ2nC/tgFp764lhtptrQTnOoeGdK597p3dWvdb+LIXp9f+qH0arvHcFlB9afynwgs12hDMIX1PxNAuho9kbMlnkZUeTl47ZbNJgDWSM+2zrg6PntX4RwSWPPAI+sqikgGNQHc6I6BBphp+kkZB0rjeLFwA+IqML9qBEPlpLCzMWGK/wrv4dVmobJseYv0LvzRYJIKk33iGtg5J96InrmYiq8g6+vGj/3UBfX0T4ZOkwK/P83HoY7iOXuzuv7YHGnXdn2shCXvoB/6TRPcHHbZyhsT+Y0/He6Wanu89nUqGh0WsTIIBNwUHuBv/dMlMUxJz89vEI/gpsqw5DsKetJ0RcFR8ll/EMF/lA1qn0jmbYu2HownJXtxw1qg8nRe60zpgUqlKrIgQC6qUNj+8vW6GEpSacdRlyxqTzY+1QJOD6FVVEF0FWsHQTD09MCcVW2DD9JBjGF69WA/Rq+vxPKjJxrH3YAz+IfuQtMfCc6zhNh7LcWoiagAs94lmEj/MIJCSDhsb1+K9n0OodGVQteWBsbhgGJXDbwU/rrDnISkOg2DNqFAfu1UI9Snb8LMfMRLgvtUfrWDM8jXNGGNwmt5nnCZz9Wi87jFb8BncktTE0KPSUDu5I8rKtr75ZiBPFTAnRA25z5vays1yO6xGlcwkgXWr9/FW29WBXQvi/EpGic+p/ARAhYoVPsDz+QrnZWGVL5pGcTgwSrB+nNKD5iSl80Icwm8eqkcKOjXErPE0DlMAyUryP7O/1GuujSjdpu/wT1WVOWYjF1NjRwuyBF711YU3fwiwJSpPcUEVcUWTrmGIvkkEx2hxKJNajp75dtWqthQLEQzaavXPTDJWK1mal0YofCaLvF8vRUfF10JLFdRMoBxLHmgC7udiVnXVLPAGYCcCk8901fZJv7+W0Quj/jpbTQU1PipK0H2iSp/0q76MdcxTfHBpmISFnh9+OyiGL9WI21WUzZtqcfvO7Clb6Grdx/8uXFpbnGUlKI8Dq0vpYx55r+Z6zpapJd95+Lewq5Wjj2u+bRC1UfsRjREhe2JSF
X-MS-Exchange-AntiSpam-MessageData-1: 
 jtMTmOuaVkUIb4Mh39hjS9d62tOxPKomQrmhbupK+tcOTHCFgZLLHpQG
X-OriginatorOrg: wolfvision.net
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 76a4449e-b62e-4c90-3089-08da2c076c62
X-MS-Exchange-CrossTenant-AuthSource: 
 HE1PR0802MB2426.eurprd08.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 02 May 2022 06:46:02.5176 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: e94ec9da-9183-471e-83b3-51baa8eb804f
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 NoyY/btp6ems/yy60LJvKX1ZKSJACQjE19lAW1LFcwxpb4A7Cwrfuv8SKsXa+QAsTiSCM4CXGtIW31K49vSXDR9dSx6ggXWazSXusEuTc4w=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: AS8PR08MB6407
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220501_234613_164375_46B86607 
X-CRM114-Status: UNSURE (   8.31  )
X-CRM114-Notice: Please train this message.
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

This USB 3.0 controller of the Rockchip RK3568 is capable of OTG/DRD
operation. Enable it in the device tree.

Signed-off-by: Michael Riesch <michael.riesch@wolfvision.net>
---
v2:
 - As pointed out by Nicolas, this is RK3568 specific. Moved change to
   rk3568.dtsi.

 arch/arm64/boot/dts/rockchip/rk3568.dtsi | 1 +
 1 file changed, 1 insertion(+)

--
2.30.2

diff --git a/arch/arm64/boot/dts/rockchip/rk3568.dtsi b/arch/arm64/boot/dts/rockchip/rk3568.dtsi
index 5eafddf62edc..bbfe8f3d68b7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568.dtsi
@@ -134,6 +134,7 @@ power-domain@RK3568_PD_PIPE {
 };

 &usb_host0_xhci {
+	dr_mode = "otg";
 	phys = <&usb2phy0_otg>, <&combphy0 PHY_TYPE_USB3>;
 	phy-names = "usb2-phy", "usb3-phy";
 };
